### ISSUE TITLE
aws-cpp-sdk-transfer fails to upload file to s3 when target bucket name contains a dot

### ISSUE THREAD

**Issue by Eliyahu-Machluf** [aws/aws-sdk-cpp#546]
Hi,

I tried to upload a file using aws-cpp-sdk-transfer project, and when the bucket name contains a dot, transfer fails. generating error:
Unable to connect to endpoint

it this by intention? this is a bug or a feature of the sdk which does not dot at bucket name?

I tried a bucket named 'test_11.this'. problem reason is that the url is being built as virtual host name.
https://test_11.this.s3.amazonaws.com/a.txt and not using the path-like url i.e. https://s3.amazonaws.com/test_11.this/a.txt

**Comment by singku** [comment]
Hi, there are some restrictions and limitations on bucket naming, the name should be DNS compliant, see [here](http://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html)

**Comment by marcomagdy** [comment]
You don't have to abide by the DNS compliance rules if you turn off virtual addressing (one of the parameters in the S3Client constructors). Virtual addressing is turned on by default.

**Comment by Eliyahu-Machluf** [comment]
Hi,

Thanks for the answers.

From the link you gave me, http://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html I understand bucket name can contain a single period. "Bucket names must be a series of one or more labels. Adjacent labels are separated by a single period (.)"
I tried turning off the 'virtual addressing' at s3 client constructor, but it did not help, and I continued getting an error "Unable to connect to endpoint"
I tried also a bucket name with period with no numbers at all (i.e. only lowercase letters), but it did not work too
